web/src/message_fetch.js

```diff
@@ -51,6 +54,17 @@ function process_result(data, opts) {
         message_util.add_old_messages(messages, opts.msg_list);
     }
 
+    if (messages.length > 0 && opts.msg_list === message_lists.home) {
+        // We keep track of how far back we've fetched messages for, for messaging in
+        // the recent view. This assumes `data.messages` is already sorted.
+        const oldest_timestamp = all_messages_data.last().timestamp;
```

Comment:
OK I'm pretty sure the display bug is that this should be doing `.first()` if you want the oldest message -- `.last()` is the most recent message :)

Reply:
Thanks!